src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -333,7 +334,8 @@ internal void ValidateTimerCallback(ActorHost host, string callback)
             }
 
             // The timer callback should return only Task return type
-            if (methodInfo.ReturnType != typeof(Task))
+            if (methodInfo.ReturnType != typeof(Task) 
+                && methodInfo.ReturnType.GetMethod("Invoke")?.ReturnType != typeof(Task)) // Support for F# 'task' computation expressions
```

Comment:
It feels like this check should be more robust. Like we should only look for `Invoke()` based on the type of `methodInfo.ReturnType`. 

What's the type that we're looking at here? Is it this? https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-fsharpfunc-2.html